Code/Framework/AzCore/AzCore/Utils/Utils.cpp

```diff
@@ -17,6 +19,44 @@
 #include <AzCore/Settings/SettingsRegistryMergeUtils.h>
 #include <AzCore/StringFunc/StringFunc.h>
 
+namespace AZ::Utils::ConsoleCommands
+{
+    // Register Console Command that can query the OS environment variables
+    constexpr const char* GetEnvCommandName = "os_GetEnv";
+    void GetEnv(const AZ::ConsoleCommandContainer& commandArgs)
+    {
+        if (commandArgs.empty())
+        {
+            AZ_Error(
+                "os_GetEnv",
+                false,
+                "No environment variable name has been supplied.");
+            return;
+        }
+
+        for (AZStd::string_view commandArg : commandArgs)
+        {
+            auto QueryOSEnvironmentVariable = [commandArg](char* buffer, size_t size)
+            {
+                AZStd::fixed_string<1024> envKey{ commandArg };
+                auto getEnvOutcome = AZ::Utils::GetEnv(AZStd::span(buffer, size), envKey.c_str());
+                return getEnvOutcome ? getEnvOutcome.GetValue().size() : 0;
+            };
+
+            AZStd::fixed_string<4096> envValue;
+            envValue.resize_and_overwrite(envValue.capacity(), QueryOSEnvironmentVariable);
+            auto formattedEnvOutput = AZStd::fixed_string<4096>::format("%.*s=%s\n", AZ_STRING_ARG(commandArg), envValue.c_str());
+            AZ::Debug::Trace::Instance().Output(GetEnvCommandName, formattedEnvOutput.c_str());
+        }
```

Comment:
Please make the 1024 and 4096 values named constants and explain where they came from (i.e. arbitrary, or "that's how big all operating systems allow environment variables to be", or whatever).

Also, shouldn't the final 4096 actually be 1024+4096+2 since it's combining fixed<1024> + "=" + fixed<4096> + "\n"?

Reply:
For the final fixed string, it should be 5120, but I stuck with 4096 to avoid a unique `AZStd::fixed_string<5120>` that doesn't exist yet in the code plus the fact that chances are that the environment variable value + the name of the environment variable would probably never hit the 4096 character limit.